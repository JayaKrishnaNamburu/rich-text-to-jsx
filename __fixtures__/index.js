/* eslint-disable max-len */
export function createDocument(content) {
  return {
    data: {},
    nodeType: 'document',
    content
  };
}

export const headingOne = {
  data: {},
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a first-level heading',
      nodeType: 'text'
    }
  ],
  nodeType: 'heading-1'
};

export const headingThree = {
  data: {},
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a third-level heading',
      nodeType: 'text'
    }
  ],
  nodeType: 'heading-3'
};

export const text = {
  data: {},
  marks: [],
  value: 'This is normal text.',
  nodeType: 'text'
};

export const bold = {
  data: {},
  marks: [
    {
      type: 'bold'
    }
  ],
  value: 'This is bold text.',
  nodeType: 'text'
};

export const italic = {
  data: {},
  marks: [
    {
      type: 'italic'
    }
  ],
  value: 'This is italic text.',
  nodeType: 'text'
};

export const underline = {
  data: {},
  marks: [
    {
      type: 'underline'
    }
  ],
  value: 'This is underlined text.',
  nodeType: 'text'
};

export const hyperlink = {
  data: {
    uri: 'https://acme.com'
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a hyperlink.',
      nodeType: 'text'
    }
  ],
  nodeType: 'hyperlink'
};

export const boldAndItalic = {
  data: {},
  marks: [
    {
      type: 'bold'
    },
    {
      type: 'italic'
    }
  ],
  value: 'This is bold and italic text.',
  nodeType: 'text'
};

export const paragraph = {
  data: {},
  content: [text, bold, italic, underline, hyperlink],
  nodeType: 'paragraph'
};

export const blockquote = {
  data: {},
  content: [paragraph],
  nodeType: 'blockquote'
};

export const listItem = {
  data: {},
  content: [paragraph],
  nodeType: 'list-item'
};

export const unorderedList = {
  data: {},
  content: [listItem, listItem, listItem],
  nodeType: 'unordered-list'
};

export const hr = {
  data: {},
  content: [],
  nodeType: 'hr'
};

const asset = {
  id: '4fgGUXCJXWOQUAEQqCS8MW',
  updatedAt: '2019-01-14T04:57:27.049Z',
  title: 'Random photo from Unsplash.com',
  file: {
    url: 'https://source.unsplash.com/random/800x500',
    details: {
      size: 240963,
      image: {
        width: 800,
        height: 500
      }
    },
    fileName: 'random-unsplash.jpg',
    contentType: 'image/jpg'
  }
};

const entry = {
  id: '32v7TZ7YQEaugOeew4SymY',
  contentType: 'page',
  updatedAt: '2019-01-18T14:37:30.221Z',
  slug: 'about',
  title: 'Title: About Acme Corp'
};

export const embeddedAsset = {
  data: {
    target: asset
  },
  content: [],
  nodeType: 'embedded-asset-block'
};

export const assetHyperlink = {
  data: {
    target: asset
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'ham hock',
      nodeType: 'text'
    }
  ],
  nodeType: 'asset-hyperlink'
};

export const entryHyperlink = {
  data: {
    target: {
      id: '32v7TZ7YQEaugOeew4SymY',
      contentType: 'route',
      updatedAt: '2019-01-18T14:37:30.221Z',
      slug: 'about'
    }
  },
  content: [
    {
      data: {},
      marks: [],
      value: 'This is a link to an entry.',
      nodeType: 'text'
    }
  ],
  nodeType: 'entry-hyperlink'
};

export const embeddedEntryInline = {
  data: {
    target: entry
  },
  content: [],
  nodeType: 'embedded-entry-inline'
};

export const embeddedEntryBlock = {
  data: {
    target: entry
  },
  content: [],
  nodeType: 'embedded-entry-block'
};
