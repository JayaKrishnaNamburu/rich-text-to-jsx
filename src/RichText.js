import PropTypes from 'prop-types';
import richTextToJsx from './RichTextService';

/**
 * Describe RichText here.
 */
const RichText = ({ children, ...options }) => richTextToJsx(children, options);

RichText.propTypes = {
  /**
   * A consice description of the example prop.
   */
  blocks: PropTypes.object,
  inlines: PropTypes.object,
  createElement: PropTypes.func
};

RichText.defaultProps = {};

/**
 * @component
 */
export default RichText;
