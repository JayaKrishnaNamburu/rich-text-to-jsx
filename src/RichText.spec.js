import React from 'react';
import { create } from 'react-test-renderer';

import { richText } from '../__fixtures__';

import RichText from './RichText';

describe('RichText', () => {
  // !TODO: write your tests.
  describe('styles', () => {
    it('should render with default styles', () => {
      const actual = create(<RichText richText={richText} />);
      expect(actual).toMatchSnapshot();
    });
  });
});
