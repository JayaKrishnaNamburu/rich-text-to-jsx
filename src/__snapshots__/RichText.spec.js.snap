// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText styles should render with default styles 1`] = `
Array [
  <h1>
    This is a first-level heading
  </h1>,
  <p>
    This is normal text.
    <strong>
      This is bold text.
    </strong>
    <em>
      This is italic text.
    </em>
    <u>
      This is underlined text.
    </u>
  </p>,
  <ul>
    <li>
      <p>
        This is normal text.
        <strong>
          This is bold text.
        </strong>
        <em>
          This is italic text.
        </em>
        <u>
          This is underlined text.
        </u>
      </p>
    </li>
    <li>
      <p>
        This is normal text.
        <strong>
          This is bold text.
        </strong>
        <em>
          This is italic text.
        </em>
        <u>
          This is underlined text.
        </u>
      </p>
    </li>
    <li>
      <p>
        This is normal text.
        <strong>
          This is bold text.
        </strong>
        <em>
          This is italic text.
        </em>
        <u>
          This is underlined text.
        </u>
      </p>
    </li>
  </ul>,
  <hr />,
  <blockquote>
    <p>
      This is normal text.
      <strong>
        This is bold text.
      </strong>
      <em>
        This is italic text.
      </em>
      <u>
        This is underlined text.
      </u>
    </p>
  </blockquote>,
]
`;
